src/components/CalendarComponents/categoriesBar.vue

```diff
@@ -26,23 +26,15 @@
 
 <script lang="ts">
 import { Component, Vue } from 'vue-property-decorator';
+
 import Category from '@/models/Category';
-import { AllCategories } from '@/db-service/Categories/queries';
 
 @Component({
-  apollo: {
-    allCategories: {
-      query: AllCategories,
-    },
+  props: {
+    categories: Array as () => Category[],
```

Comment:
There is a @Prop annotation for properties, https://github.com/kaorun343/vue-property-decorator/issues/231